spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestDataSourceOptions.java

```diff
@@ -404,4 +414,61 @@ public void testExtraSnapshotMetadata() throws IOException {
     Assert.assertTrue(table.currentSnapshot().summary().get("extra-key").equals("someValue"));
     Assert.assertTrue(table.currentSnapshot().summary().get("another-key").equals("anotherValue"));
   }
+
+  @Test
+  public void testExtraSnapshotMetadataWithSQL() throws IOException {
+    String tableLocation = temp.newFolder("iceberg-table").toString();
+    HadoopTables tables = new HadoopTables(CONF);
+    int threadsCount = 3;
+    ExecutorService executorService = Executors.newFixedThreadPool(threadsCount, new ThreadFactory() {
```

Comment:
Can you use the helper in `ThreadPools` and also wrap it in a try/finally to close the threadpool?

I'm also not entirely sure this requires a threadpool to test. I think it would be fine to test a single write in the current thread.

Reply:
I would also prefer a test that uses a single write in the current thread without any additional threading business. I worry that `CommitMetadata` doesn't _seem_ thread local to users.

And then if a multi-threaded test is needed, using the helpers from `ThreadPools` as suggested.

Reply:
updated to ThreadPool, I think multi-threading testing is still necessary? as we need to have something guarding that the commit metadata change is thread safe no matter we use ThreadLocal as now or later we change to something else for any reason

Reply:
I don't think multi-threaded testing is needed. It's enough to know that we're using a thread-local. This also is not guaranteed to run the way this test assumes that it will. There is not a guarantee that the thread pool will scale all the way up, and there's no guarantee that the tasks will each run in a separate thread. I think it's likely that those will happen, but this could still be a source of flakiness later on. Also, this doesn't necessarily test that the thread-local is working properly because there's no guarantee of concurrency across tasks.

While it's probably working the way you expect, there's no guarantee that it must. So I'd prefer to keep the test simple.

Reply:
updated